src/importlinter/adapters/user_options.py

```diff
@@ -51,3 +61,38 @@ def _clean_section_config(section_config: Dict[str, Any]) -> Dict[str, Any]:
             else:
                 section_dict[key] = value.strip().split("\n")
         return section_dict
+
+
+class TomlUserOptionReader(ports.UserOptionReader):
+    """
+    Reader that looks for and parses the contents of TOML files.
+    """
+
+    section_name = "importlinter"
+
+    def read_options(self, config_filename: Optional[str] = None) -> Optional[UserOptions]:
+        if not _HAS_TOML:
+            return None
+
+        if config_filename:
```

Comment:
NIP: to avoid `if else` statement you can use `or` operator:

```python
config_filename = config_filename or 'pyproject.toml'
```

or even set `pyproject.toml` as `config_filename` parameter default value.

Reply:
With this approach it's not possible to fail when the user provides a non-existing file. So the current logic is, that if a user specifies a file, the file must exist. Else a non-existing file is silently ignored